packages/react-vapor/src/components/editor/tests/CodeEditor.spec.tsx

```diff
@@ -133,5 +111,18 @@ describe('CodeEditor', () => {
         it('should have a border by default', () => {
             expect(codeEditor.find(ReactCodeMirror.Controlled).props().className).toContain('mod-border');
         });
+
+        it('should modify the store on change if mounted with an Id', () => {
+            const store = getStoreMock();
+
+            const component = mountWithStore(<CodeEditor id="anId" value="a value" />, store);
+
+            const inputArea = component.find('div');
+
+            inputArea.simulate('change', {target: {value: 'new value'}});
+
+            expect(store.getActions().length).toBe(1);
+            expect(store.getActions()[0].type).toBe(CodeEditorActionTypes.update);
+        });

```

Comment:
Really wanted to use RTL. But couldn't find a way to get the dispatched store actions

Reply:
This is an implementation detail, that's why its hard to test it with RTL

Reply:
Checking the dispatching of the actions might be an implementation detail, but I dont see another way I could test that the component interacts with the store properly.

Reply:
does the modify store reflect in any way in the ui?

Reply:
no. the behavior is the same. But imo its important to know the component is dispatching actions

Reply:
One way to test it would be to create a fixture component conencted to the store that uses the code editor value and render that component next to the CodeEditor in the RTL `render()`. For example a button with an onclick handler. Then in your test you could interact with that fixture component (for example click on the button) to assert that the code editor value is the correct one.

Reply:
Isn't that a bit extreme when the reducer and the selector are both tested independently ?.  

I know the confidence in the tests are now the new thing and I agree, but making sure the component is dispatching the actions wouldn't break unless someone refactors the component and changes the action names or doesn't call the actions at all. So IMO just having the capacity to access the store dispatched actions while implenting tests with RTL would give us the capacity to create tests based on the interaction of the component with the store without being flaky. Since the reducers and the selectors must be tested anyway.  

I know that this won't be the case in React vapor, but what if a component has to send a fetch with a thunkAction. I think that having the possibility to access the actions dispatched would be very helpfull then.

Reply:
I'm not saying that we should absolutely use that approach, I'm just explaining what would be the correct way to test it out using the react-testing-library approach. 

Another aspect to consider as well with that approach is that the reducers, actions and selectors wouldn't have to be tested independently because those tests would be redundant. Everything would already be covered by the CodeEditor's tests. In a full 100% RTL mindset, only the CodeEditor component functionality would be tested from a `react-vapor` library user standpoint. That means no store mock, no watching on actions being dispatched, and no assertions on the state directly.

Reply:
Yes I agree that by using this approach. All the tests we do for the reducers and selectors and so on wouldn't be that useful.

 I am willing to chat about that if we intent to completely ditch the black box concept of our test. I feel that a quick strategies sharing and brainstorm could help us save some time in the future.